### ISSUE TITLE
[Bug]: Miru v5 does not launch on linux

### ISSUE THREAD

**Issue by rainfallmelonflysong** [ThaUnknown/miru#436]
### Preflight checklist

- [X] I have searched the [issue tracker](https://github.com/ThaUnknown/miru/issues) for a bug report that matches the one I want to file, without success.
- [X] I have searched the [frequently asked questions](https://github.com/ThaUnknown/miru/blob/master/docs/faq.md) for a solution that fixes this problem, without success.
- [X] I have checked that I'm using the [latest stable](https://github.com/ThaUnknown/miru/releases/latest) version of the app.

### What app version are you using?

5.0.2

### What operating system version are you using?

Linux

### Expected Behavior

Miru launches with no issues

### Actual Behavior

When attempting to launch miru v5 on linux, the app displays a blank screen for 3 secs and then crashes, giving this (electron?) error

`
Checking for beta autoupdate feature for deb/rpm distributions
Found package-type: deb
19:20:59.399 › Checking for update
[4595:0325/192100.022102:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[4595:0325/192100.505817:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
19:21:00.564 › Update for version 5.0.2 is not available (latest version: 4.5.10, downgrade is disallowed).
19:21:00.564 › checkForUpdatesAndNotify called, downloadPromise is null
[4595:0325/192101.030652:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[4595:0325/192101.658375:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[4595:0325/192102.281327:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[4595:0325/192102.887335:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[4595:0325/192103.176312:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[4595:0325/192103.466594:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[4595:0325/192103.763053:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[4595:0325/192103.763068:FATAL:gpu_data_manager_impl_private.cc(448)] GPU process isn't usable. Goodbye.
[1]    4595 trace trap (core dumped)  miru`

**Comment by Muril-o** [comment]
Need testing, but it's probably electron issue, maybe related to one of the flags set [here](https://github.com/ThaUnknown/miru/blob/92e5207e3a5a1c5dc2510b2e9cd88bd041391d57/electron/src/main/util.js#L6-L19).
I cannot reproduce this on wayland+nvidia proprietary 550.54.14 drivers, would help sending more info on GPU, current driver, distro, window manager, display server, etc.
The relevant output of `inxi -Fxxrzc0` should be enough, together with the method you used to install (AppImage or deb) to help isolate the problem.
Meanwhile, from discord discussion, running miru with `miru --in-process-gpu` should work, tho I have no clue what implications this flag has on actual usage.

**Comment by rainfallmelonflysong** [comment]
Indeed, it seems to work by using the `--in-process-gpu` flag. I'm running arch with x11/i3/nvidia-dkms with no compositor, so not technically supported. I've also tried running the appimage with the same result.

**Comment by ThaUnknown** [comment]
`no-sandbox` likely causes this, but since we've got no reproduction this is being labeled as a "linux config issue"

**Comment by CaptainD54321** [comment]
Can replicate this issue on my linux machine, crashes with identical errors as above unless I launch with the `--in-process-gpu` flag, here's output from `inxi -Fxxrzc0`:
<details>
<summary> Output </summary>

```
System:
  Kernel: 5.15.0-101-generic x86_64 bits: 64 compiler: gcc v: 11.4.0
    Desktop: Cinnamon 6.0.4 tk: GTK 3.24.33 wm: muffin dm: LightDM
    Distro: Linux Mint 21.3 Virginia base: Ubuntu 22.04 jammy
Machine:
  Type: Laptop System: Alienware product: Alienware m15 v: 2.12.0
    serial: <superuser required> Chassis: type: 10 serial: <superuser required>
  Mobo: Alienware model: 01F1MV v: A00 serial: <superuser required>
    UEFI: Alienware v: 2.12.0 date: 01/11/2022
CPU:
  Info: 6-core model: Intel Core i7-8750H bits: 64 type: MT MCP
    arch: Coffee Lake rev: A cache: L1: 384 KiB L2: 1.5 MiB L3: 9 MiB
  Speed (MHz): avg: 3870 high: 3986 min/max: 800/4100 cores: 1: 3841
    2: 3932 3: 3964 4: 3843 5: 3902 6: 3802 7: 3986 8: 3816 9: 3938 10: 3810
    11: 3856 12: 3752 bogomips: 52799
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx
Graphics:
  Device-1: Intel CoffeeLake-H GT2 [UHD Graphics 630] vendor: Dell
    driver: i915 v: kernel ports: active: eDP-1
    empty: DP-1, DP-2, HDMI-A-1, HDMI-A-2 bus-ID: 00:02.0 chip-ID: 8086:3e9b
  Device-2: NVIDIA TU106M [GeForce RTX 2060 Mobile] vendor: Dell
    driver: nvidia v: 525.147.05 pcie: speed: 2.5 GT/s lanes: 8 ports:
    active: none off: HDMI-A-3 empty: DP-3 bus-ID: 01:00.0 chip-ID: 10de:1f11
  Device-3: Sunplus Innovation Integrated_Webcam_FHD type: USB
    driver: uvcvideo bus-ID: 1-9:8 chip-ID: 1bcf:2b93
  Display: x11 server: X.Org v: 1.21.1.4 driver: X:
    loaded: modesetting,nvidia unloaded: fbdev,nouveau,vesa gpu: i915
    display-ID: :0 screens: 1
  Screen-1: 0 s-res: 3840x1080 s-dpi: 96
  Monitor-1: HDMI-0 pos: primary,right res: 1900x1068 dpi: 84
    diag: 660mm (26")
  Monitor-2: eDP-1-1 pos: primary,left res: 1920x1080 dpi: 142
    diag: 395mm (15.5")
  OpenGL: renderer: N/A v: N/A direct render: N/A
Audio:
  Device-1: Intel Cannon Lake PCH cAVS vendor: Dell driver: snd_hda_intel
    v: kernel bus-ID: 00:1f.3 chip-ID: 8086:a348
  Device-2: NVIDIA TU106 High Definition Audio vendor: Dell
    driver: snd_hda_intel v: kernel pcie: speed: 2.5 GT/s lanes: 8
    bus-ID: 01:00.1 chip-ID: 10de:10f9
  Sound Server-1: ALSA v: k5.15.0-101-generic running: yes
  Sound Server-2: PulseAudio v: 15.99.1 running: yes
  Sound Server-3: PipeWire v: 0.3.48 running: yes
Network:
  Device-1: Qualcomm Atheros Killer E2500 Gigabit Ethernet
    vendor: Rivet Networks driver: alx v: kernel pcie: speed: 2.5 GT/s lanes: 1
    port: 3000 bus-ID: 6f:00.0 chip-ID: 1969:e0b1
  IF: enp111s0 state: down mac: <filter>
  Device-2: Intel Wireless-AC 9260 vendor: Rivet Networks driver: iwlwifi
    v: kernel pcie: speed: 5 GT/s lanes: 1 bus-ID: 70:00.0 chip-ID: 8086:2526
  IF: wlp112s0 state: up mac: <filter>
  IF-ID-1: wg0-mullvad state: unknown speed: N/A duplex: N/A mac: N/A
Bluetooth:
  Device-1: Intel Wireless-AC 9260 Bluetooth Adapter type: USB driver: btusb
    v: 0.8 bus-ID: 1-7:6 chip-ID: 8087:0025
  Report: hciconfig ID: hci0 rfk-id: 2 state: up address: <filter>
    bt-v: 3.0 lmp-v: 5.1 sub-v: 100
Drives:
  Local Storage: total: 5.01 TiB used: 4.98 TiB (99.2%)
  ID-1: /dev/nvme0n1 vendor: Toshiba model: KXG60ZNV512G NVMe 512GB
    size: 476.94 GiB speed: 31.6 Gb/s lanes: 4 serial: <filter> temp: 49.9 C
  ID-2: /dev/sda type: USB vendor: Western Digital model: WD50NPZZ-00A9JT0
    size: 4.55 TiB serial: <filter>
Partition:
  ID-1: / size: 452.89 GiB used: 415.01 GiB (91.6%) fs: ext4
    dev: /dev/nvme0n1p3
  ID-2: /boot/efi size: 486 MiB used: 6.1 MiB (1.3%) fs: vfat
    dev: /dev/nvme0n1p1
Swap:
  ID-1: swap-1 type: partition size: 15.26 GiB used: 213.8 MiB (1.4%)
    priority: -2 dev: /dev/nvme0n1p2
Sensors:
  System Temperatures: cpu: 93.0 C pch: 74.0 C mobo: 27.8 C gpu: nvidia
    temp: 64 C
  Fan Speeds (RPM): N/A
Repos:
  Packages: 3039 apt: 3023 flatpak: 16
  No active apt repos in: /etc/apt/sources.list
  Active apt repos in: /etc/apt/sources.list.d/ksp-ckan.list
    1: deb [arch=amd64 signed-by=/usr/share/keyrings/ksp-ckan-archive-keyring.gpg] https://ksp-ckan.s3-us-west-2.amazonaws.com/deb stable main
  Active apt repos in: /etc/apt/sources.list.d/lutris.list
    1: deb https://download.opensuse.org/repositories/home:/strycore/Debian_11/ ./
  Active apt repos in: /etc/apt/sources.list.d/mullvad.list
    1: deb [signed-by=/usr/share/keyrings/mullvad-keyring.asc arch=amd64] https://repository.mullvad.net/deb/stable virginia main
  Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list
    1: deb http://packages.linuxmint.com virginia main upstream import backport
    2: deb http://archive.ubuntu.com/ubuntu jammy main restricted universe multiverse
    3: deb http://archive.ubuntu.com/ubuntu jammy-updates main restricted universe multiverse
    4: deb http://archive.ubuntu.com/ubuntu jammy-backports main restricted universe multiverse
    5: deb http://security.ubuntu.com/ubuntu/ jammy-security main restricted universe multiverse
  No active apt repos in: /etc/apt/sources.list.d/steam-beta.list
  Active apt repos in: /etc/apt/sources.list.d/steam-stable.list
    1: deb [arch=amd64,i386 signed-by=/usr/share/keyrings/steam.gpg] https://repo.steampowered.com/steam/ stable steam
    2: deb-src [arch=amd64,i386 signed-by=/usr/share/keyrings/steam.gpg] https://repo.steampowered.com/steam/ stable steam
  Active apt repos in: /etc/apt/sources.list.d/unityhub.list
    1: deb [signed-by=/usr/share/keyrings/Unity_Technologies_ApS.gpg] https://hub.unity3d.com/linux/repos/deb stable main
  Active apt repos in: /etc/apt/sources.list.d/vscode.list
    1: deb [arch=amd64,arm64,armhf] http://packages.microsoft.com/repos/code stable main
  Active apt repos in: /etc/apt/sources.list.d/winehq-bullseye.sources
    1: deb [arch=amd64 i386] https://dl.winehq.org/wine-builds/debian bullseye main
  Active apt repos in: /etc/apt/sources.list.d/winehq-jammy.sources
    1: deb [arch=amd64 i386] https://dl.winehq.org/wine-builds/ubuntu jammy main
Info:
  Processes: 461 Uptime: 23h 42m Memory: 31.14 GiB used: 14.15 GiB (45.4%)
  Init: systemd v: 249 runlevel: 5 Compilers: gcc: 11.4.0 alt: 11/12
  Shell: Bash v: 5.1.16 running-in: gnome-terminal inxi: 3.3.13
```
</details>

**Comment by VictorGamerLOL** [comment]
This also occurs on my arch system. Some digging later I found this might be an electron issue or possibly even chromium itself. I ensured I have all of the appropriate libva drivers present on my AMD integrated graphics system with no avail. `--in-process-gpu` was the only one that stopped the crashing.

**Comment by ThaUnknown** [comment]
@VictorGamerLOL @CaptainD54321 @rainfallmelonflysong I've released https://github.com/ThaUnknown/miru/releases/tag/v5.0.7 which is a beta, download and install manually, check if issue persists

**Comment by rainfallmelonflysong** [comment]
@ThaUnknown Tried 5.0.7 and the issue sadly persists (same behaviour and same --in-process-gpu workaround), same with 5.0.10

**Comment by VictorGamerLOL** [comment]
@ThaUnknown works for me. The activities I usually do that induce the crashing, like getting the popup to choose a torrent, no longer do.

**Comment by VictorGamerLOL** [comment]
> @ThaUnknown works for me. The activities I usually do that induce the crashing, like getting the popup to choose a torrent, no longer do. And this was 5.0.7 not 5.0.10. I am using the Appimage version.
> 
> Some more info if it is of any relevance:
> 
> pacman -Q | grep libva:
> 
> ```
> lib32-libva 2.20.0-1
> libva 2.21.0-1
> libva-mesa-driver 1:24.0.5-1
> libva-utils 2.21.0-1
> ```
> 
> pacman -Q | grep mesa:
> 
> ```
> lib32-mesa 1:24.0.5-1
> libva-mesa-driver 1:24.0.5-1
> mesa 1:24.0.5-1
> mesa-utils 9.0.0-4
> mesa-vdpau 1:24.0.5-1
> opencl-clover-mesa 1:24.0.5-1
> opencl-rusticl-mesa 1:24.0.5-1
> ```
> 
> lspci | grep VGA:
> 
> ```
> 04:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Lucienne (rev c1)
> ```
> 
> Edit: 5.0.10 also does not crash.

I retract my words. Aggressively scrolling still induces a `GPU process exited unexpectedly: exit_code=133` but not enough for it to be considered unusable and crash the entire app. Still stands.

**Comment by CaptainD54321** [comment]
Huh I swore I posted this reply the other day, but apparently github ate it: @ThaUnknown I tried both 5.0.7 and the latest version (5.0.12) and my issue (Miru crashing immediately on launch) persists if I don't use the `--in-process-gpu` flag. 
<details>
<summary><b>Log output from Miru:</b></summary>
```
Checking for beta autoupdate feature for deb/rpm distributions
Found package-type: deb
15:29:43.510 › Checking for update
[13960:0419/152944.631384:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
15:29:44.670 › Update for version 5.0.12 is not available (latest version: 5.0.12, downgrade is disallowed).
15:29:44.671 › checkForUpdatesAndNotify called, downloadPromise is null
[13960:0419/152945.558397:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[13960:0419/152946.435602:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[13960:0419/152947.230013:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[13960:0419/152947.701148:ERROR:quota_database.cc(936)] Could not open the quota database, resetting.
[13960:0419/152947.990075:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[13960:0419/152948.809085:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[13960:0419/152949.218631:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[13960:0419/152949.601394:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[13960:0419/152949.983976:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=133
[13960:0419/152949.983994:FATAL:gpu_data_manager_impl_private.cc(448)] GPU process isn't usable. Goodbye.
Trace/breakpoint trap (core dumped)
```
</summary>
